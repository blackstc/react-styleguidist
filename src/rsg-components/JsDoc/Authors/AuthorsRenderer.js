import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'rsg-components/Styled';
import Markdown from 'rsg-components/Markdown';

export const styles = ({ font }) => ({
	authorsBlock: {
		marginTop: 8,
		fontSize: 'inherit',
	},
	authorsHeader: {
		fontStyle: 'italic',
		fontSize: 'inherit',
		fontFamily: font,
	},
	authorsList: {
		marginTop: 3,
		marginBottom: 3,
		marginLeft: 20,
	},
});

export function AuthorsRenderer({ classes, tags }) {
	if (!tags || !tags.author) {
		return null;
	}

	function renderAuthors(authors) {
		return (
			authors.map((author, index) => {
				return (
					<li key={`author-${index}`}>
						<Markdown text={author.description} inline />
					</li>
				);
			})
		);
	}

	return (
		<div className={classes.authorsBlock}>
			<div className={classes.authorsHeader}>{tags.author.length === 1 ? 'Author' : 'Authors'}</div>
			<ul className={classes.authorsList}>
				{ renderAuthors(tags.author) }
			</ul>
		</div>
	);
}

AuthorsRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
	tags: PropTypes.shape({
		author: PropTypes.array,
	}).isRequired,
};

export default Styled(styles)(AuthorsRenderer);
