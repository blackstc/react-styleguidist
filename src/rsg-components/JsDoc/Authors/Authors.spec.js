import React from 'react';
import { AuthorsRenderer } from './AuthorsRenderer';

describe('Authors', () => {
	const authorOne = {
		title: 'author',
		description: 'Test author',
	};
	const authorTwo = {
		title: 'author',
		description: '[Test](#TestLink)',
	};

	it('renderer should render for one @author tag', () => {
		const tags = {
			author: [authorOne],
		};

		const actual = render(
			<AuthorsRenderer classes={{}} tags={tags} />
		);

		expect(actual).toMatchSnapshot();
	});

	it('renderer should render for two @author tags', () => {
		const tags = {
			author: [authorOne, authorTwo],
		};

		const actual = render(
			<AuthorsRenderer classes={{}} tags={tags} />
		);

		expect(actual).toMatchSnapshot();
	});

	it('renderer should not render for empty tags', () => {
		const actual = shallow(
			<AuthorsRenderer classes={{}} tags={{}} />
		);

		expect(actual.node).toBe(null);
	});
});

