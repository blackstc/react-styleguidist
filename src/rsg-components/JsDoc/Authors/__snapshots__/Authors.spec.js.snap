// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authors renderer should render for one @author tag 1`] = `
<div>
  <div>
    Author
  </div>
  <ul>
    <li>
      <span
        class="base-0-0"
      >
        Test author
      </span>
    </li>
  </ul>
</div>
`;

exports[`Authors renderer should render for two @author tags 1`] = `
<div>
  <div>
    Authors
  </div>
  <ul>
    <li>
      <span
        class="base-0-0"
      >
        Test author
      </span>
    </li>
    <li>
      <span
        class="base-0-0"
      >
        <a
          class="a-0-2"
          href="#TestLink"
        >
          Test
        </a>
      </span>
    </li>
  </ul>
</div>
`;
